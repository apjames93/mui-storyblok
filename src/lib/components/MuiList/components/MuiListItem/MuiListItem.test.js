import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiListItem from './MuiListItem';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiListItem", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    listItemAvatar: [{
      _editable: '<!--#storyblok#{"name": "MuiListItemAvatar", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiListItemAvatar',
      content: [{
        _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],

    listItemIcon: [{
      _editable: '<!--#storyblok#{"name": "MuiListItemIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiListItemIcon',
      content: [{
        _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],

    listItemText: [{
      _editable: '<!--#storyblok#{"name": "MuiListItemText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiListItemText',
      primary: 'primary',
    }],

    listItemSecondaryAction: [{
      _editable: '<!--#storyblok#{"name": "MuiListItemSecondaryAction", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiListItemSecondaryAction',
      content: [{
        _editable: '<!--#storyblok#{"name": "MuiIconButtonRedirect", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiIconButtonRedirect',
        iconButton: [{
          _editable: '<!--#storyblok#{"name": "MuiIconButton", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
          component: 'MuiIconButton',
          icon: [{
            _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
            component: 'MuiIcon',
            iconName: 'star',
          }],
        }],
      }],
    }],
  };
  const comp = shallow(<MuiListItem {...props} />);
  return { comp, props };
}

describe('<MuiListItem />', () => {
  it('renders MuiListItem', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiListItem {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
