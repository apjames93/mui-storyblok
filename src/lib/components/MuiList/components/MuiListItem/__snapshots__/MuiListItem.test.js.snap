// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiListItem /> snapshot 1`] = `
<li
  className="MuiListItem-root makeStyles-root-3 MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root makeStyles-root-4"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-5 MuiIcon-colorSecondary"
      >
        android
      </span>
    </div>
  </div>
  <div
    className="MuiListItemIcon-root makeStyles-root-6"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-7 MuiIcon-colorSecondary"
    >
      android
    </span>
  </div>
  <div
    className="MuiListItemText-root makeStyles-root-8 MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      primary
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      
    </p>
  </div>
  <div
    className="MuiListItemSecondaryAction-root makeStyles-root-9"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root makeStyles-root-10"
      data-testid="muiIconButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root makeStyles-root-11 MuiIcon-colorSecondary"
        >
          star
        </span>
      </span>
    </button>
  </div>
</li>
`;
